### ISSUE TITLE
Callback mechanism in Trainer

### ISSUE THREAD

**Issue by lostella** [awslabs/gluon-ts#706]
As suggested in another thread, it would be nice to add a callback mechanism to the Trainer, so that one can, e.g., provide custom learning rate scheduling or stopping criteria, or do custom logging. This could be addressed as part of a more general refactoring of the Trainer class, which currently is not very flexible.

A discussion and a few suggestions are in #555.

**Comment by sunny1401** [comment]
Hi - I have been trying to work party on this issue to add logging to deepar - for getting training and validation loss. I just want to confirm - I am collecting the losses in trainer._base - __call__ method - just before they are returned from the loop function. Do you think that is right way to go here?
